numpy/f2py/tests/test_callback.py

```diff
@@ -133,6 +144,25 @@ def callback(code):
         r = f(callback)
         assert_(r == 0, repr(r))
 
+    @dec.knownfailureif(sys.platform=='win32',
+                        msg='Fails with MinGW64 Gfortran (Issue #9673)')
+    def test_string_callback_array(self):
+        # See gh-10027
+        cu = np.zeros((1, 8), 'S1')
+
+        def callback(cu, lencu):
+            if cu.shape != (lencu, 8):
+                return 1
+            if cu.dtype != 'S1':
+                return 2
+            if not np.all(cu == b''):
+                return 3
+            return 0
```

Comment:
I'm curious - what happens if you use `assert` in here directly?

Reply:
Don't know, I was tempted but figured to play safe ... <tries> Yes, that works.  The fortran subroutine doesn't catch the error, rather the call wrapper does a `longjmp` to an error return from the fortran subroutine C wrapper.